### PR TITLE
Http source otel logs source test coverage

### PR DESCRIPTION
**Pull request by Galactus22625**
### Description
[Describe what this change achieves]
 Adding additional test coverage for http source and otel logs source
### Issues Resolved
Resolves #[Issue number to be closed when this PR is merged]
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java [hunk 1]
```diff
@@ -603,6 +617,82 @@ public void testServerStartCertFileSuccess() throws IOException {
...
+            when(sourceConfig.getSslCertificateFile()).thenReturn(null);
+            when(sourceConfig.getSslKeyFile()).thenReturn(null);
+            HTTPSourceUnderTest = new HTTPSource(sourceConfig, pluginMetrics, pluginFactory, pipelineDescription);
+            assertThrows(NullPointerException.class, () -> HTTPSourceUnderTest.start(testBuffer));
+        }
+    }
+
+    @Test
+    void testServerStartACMCertSuccess() throws IOException {
+        try (MockedStatic<Server> armeriaServerMock = Mockito.mockStatic(Server.class)) {
```
Comment (dlvenable) [inline data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java:635]: Please try to reduce the amount of code in the `try` block for the `mockStatic`. I suspect you need it only when calling `HTTPSourceUnderTest.start`.

```
HTTPSourceUnderTest = new HTTPSource(sourceConfig, pluginMetrics, pluginFactory, pipelineDescription);

Field field = HTTPSourceUnderTest.getClass().getDeclaredField("certificateProviderFactory");
field.setAccessible(true);
field.set(HTTPSourceUnderTest, certificateProviderFactory);

try (MockedStatic<Server> armeriaServerMock = Mockito.mockStatic(Server.class)) {
   HTTPSourceUnderTest.start(testBuffer);
}

HTTPSourceUnderTest.stop();
```

File: data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java [hunk 2]
```diff
@@ -603,6 +617,82 @@ public void testServerStartCertFileSuccess() throws IOException {
         }
     }
 
+    @Test
+    public void testServerStartCertFileMissing() {
+        try (MockedStatic<Server> armeriaServerMock = Mockito.mockStatic(Server.class)) {
```
Comment (dlvenable) [inline data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java:622]: Similar comment: reduce the amount of code in this `try` block.

File: data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java [hunk 3]
```diff
@@ -603,6 +617,82 @@ public void testServerStartCertFileSuccess() throws IOException {
...
+            assertThat(actualCertificate, is(certAsString));
+            assertThat(actualPrivateKey, is(keyAsString));
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test
+    void testServerStartACMCertNull() {
+        try (MockedStatic<Server> armeriaServerMock = Mockito.mockStatic(Server.class)) {
```
Comment (dlvenable) [inline data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java:674]: Similar comment: reduce the amount of code in this `try` block.

File: data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java [hunk 4]
```diff
@@ -669,13 +759,13 @@ public void testDoubleStart() {
         // starting server
         HTTPSourceUnderTest.start(testBuffer);
         // double start server
-        Assertions.assertThrows(IllegalStateException.class, () -> HTTPSourceUnderTest.start(testBuffer));
+        assertThrows(IllegalStateException.class, () -> HTTPSourceUnderTest.start(testBuffer));
```
Comment (dlvenable) [inline data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java:762]: Thank you for this clean up!

File: data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java [hunk 5]
```diff
@@ -603,6 +617,82 @@ public void testServerStartCertFileSuccess() throws IOException {
...
+            HTTPSourceUnderTest.stop();
+
+            final ArgumentCaptor<InputStream> certificateIs = ArgumentCaptor.forClass(InputStream.class);
+            final ArgumentCaptor<InputStream> privateKeyIs = ArgumentCaptor.forClass(InputStream.class);
+            verify(serverBuilder).tls(certificateIs.capture(), privateKeyIs.capture());
+            final String actualCertificate = IOUtils.toString(certificateIs.getValue(), StandardCharsets.UTF_8.name());
+            final String actualPrivateKey = IOUtils.toString(privateKeyIs.getValue(), StandardCharsets.UTF_8.name());
+            assertThat(actualCertificate, is(certAsString));
+            assertThat(actualPrivateKey, is(keyAsString));
+        } catch (NoSuchFieldException | IllegalAccessException e) {
```
Comment (dlvenable) [inline data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java:667]: Do not catch exceptions in tests. For any checked exceptions, just add the `throws` block to the function name.

```
void testServerStartACMCertSuccess() throws IOException, ReflectiveOperationException
```

File: data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java [hunk 6]
```diff
@@ -603,6 +617,82 @@ public void testServerStartCertFileSuccess() throws IOException {
...
+            when(sourceConfig.isSsl()).thenReturn(true);
+
+            HTTPSourceUnderTest = new HTTPSource(sourceConfig, pluginMetrics, pluginFactory, pipelineDescription);
+
+            Field field = HTTPSourceUnderTest.getClass().getDeclaredField("certificateProviderFactory");
+            field.setAccessible(true);
+            field.set(HTTPSourceUnderTest, certificateProviderFactory);
+
+            assertThrows(NullPointerException.class, () -> HTTPSourceUnderTest.start(testBuffer));
+        } catch (NoSuchFieldException | IllegalAccessException e) {
```
Comment (dlvenable) [inline data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java:689]: See comments above about not catching exceptions.

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java (hunk 2 above)
Comment (Galactus22625) [reply data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java:622]: 🫡

File: data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java (hunk 1 above)
Comment (Galactus22625) [reply data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java:635]: 🫡

File: data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java (hunk 5 above)
Comment (Galactus22625) [reply data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java:667]: 🫡

File: data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java (hunk 6 above)
Comment (Galactus22625) [reply data-prepper-plugins/http-source/src/test/java/org/opensearch/dataprepper/plugins/source/loghttp/HTTPSourceTest.java:689]: 🫡

File: data-prepper-plugins/otel-logs-source/src/test/java/org/opensearch/dataprepper/plugins/source/otellogs/OTelLogsSourceTest.java [hunk 7]
```diff
@@ -770,6 +770,40 @@ void gRPC_with_auth_request_writes_to_buffer_with_successful_response() throws E
         assertThat(actualBufferWrites, hasSize(1));
     }
 
+    @Test
+    void gRPC_with_auth_request_with_invalid_basic_auth_does_not_write_to_buffer() throws Exception{
```
[the remark this answers is not in the document]
Comment (Galactus22625) [reply data-prepper-plugins/otel-logs-source/src/test/java/org/opensearch/dataprepper/plugins/source/otellogs/OTelLogsSourceTest.java:774]: renamed

File: data-prepper-plugins/otel-logs-source/src/test/java/org/opensearch/dataprepper/plugins/source/otellogs/OTelLogsSourceTest.java [hunk 8]
```diff
@@ -770,6 +770,40 @@ void gRPC_with_auth_request_writes_to_buffer_with_successful_response() throws E
...
+        when(httpBasicAuthenticationConfig.getUsername()).thenReturn(USERNAME);
+        when(httpBasicAuthenticationConfig.getPassword()).thenReturn(PASSWORD);
+        final GrpcAuthenticationProvider grpcAuthenticationProvider = new GrpcBasicAuthenticationProvider(httpBasicAuthenticationConfig);
+
+        when(pluginFactory.loadPlugin(eq(GrpcAuthenticationProvider.class), any(PluginSetting.class)))
+                .thenReturn(grpcAuthenticationProvider);
+        when(oTelLogsSourceConfig.enableUnframedRequests()).thenReturn(true);
+        when(oTelLogsSourceConfig.getAuthentication()).thenReturn(new PluginModel("http_basic",
+                Map.of(
+                        "username", "wrong Username",
```
[the remark this answers is not in the document]
Comment (Galactus22625) [reply data-prepper-plugins/otel-logs-source/src/test/java/org/opensearch/dataprepper/plugins/source/otellogs/OTelLogsSourceTest.java:784]: oh I didn't need to do that.  Only needed to change it in the client builder.  changing back to USERNAME, PASSWORD

**Comment by Galactus22625** [comment]
will commit changes from comments and a few additional tests soon

**Review by chenqi0805 — APPROVED** [review]

**Pull request merged**